### PR TITLE
Migrate to golangci-lint

### PR DESCRIPTION
**Pull request by sayboras**
To fix https://github.com/docker/distribution/issues/2944

Currently, I keep the same set of linters we have from gometalinter. 

Minor changes for below linting error
```
cmd/registry-api-descriptor-template/main.go:24                 goimports  File is not `goimports`-ed
registry/storage/driver/middleware/cloudfront/middleware.go:19  goimports  File is not `goimports`-ed
blobs.go:13                                                     goimports  File is not `goimports`-ed
```

### CODE REVIEW

**Comment by sayboras** [comment]
@tariq1890 please help to take a look on this PR. Thanks

File: Makefile [hunk 1]
```diff
@@ -50,7 +50,7 @@ version/version.go:
 
 check: ## run all linters (TODO: enable "unused", "varcheck", "ineffassign", "unconvert", "staticheck", "goimports", "structcheck")
 	@echo "$(WHALE) $@"
-	gometalinter --config .gometalinter.json ./...
+	@golangci-lint run
```
Comment (cappyzawa) [inline Makefile]:
```suggestion
	@GO111MODULE=off golangci-lint run
```

File: registry/registry.go [hunk 2]
```diff
@@ -12,7 +12,6 @@ import (
 	"syscall"
 	"time"
 
-	"github.com/Shopify/logrus-bugsnag"
```
Comment (cappyzawa) [inline registry/registry.go]:
```suggestion
	logrus_bugsnag "github.com/Shopify/logrus-bugsnag"
```

Comment (sayboras) [reply registry/registry.go]: thanks a lot :100:

**Comment by sayboras** [comment]
@dmcgowan @manishtomar @tariq1890 any input for this PR ? Thanks

**Comment by dmcgowan** [comment]
@sayboras I'll take a look

File: .golangci.yml [hunk 3]
```diff
@@ -0,0 +1,150 @@
+# options for analysis running
```
Comment (dmcgowan) [inline .golangci.yml]: Where did this version of the lint config come from?

In general, I would like to see this converge with containerd's configuration, see https://github.com/containerd/containerd/blob/master/.golangci.yml

If there are changes here I would like to see them applied in both places

File: go.mod [hunk 4]
```diff
@@ -7,51 +7,48 @@ require (
...
 	golang.org/x/oauth2 v0.0.0-20190604053449-0f29369cfe45
-	golang.org/x/sys v0.0.0-20190602015325-4c4f7f33c9ed // indirect
+	golang.org/x/sys v0.0.0-20191112214154-59a1497f0cea // indirect
+	golang.org/x/text v0.3.2 // indirect
 	google.golang.org/api v0.0.0-20160322025152-9bf6e6e569ff
 	google.golang.org/cloud v0.0.0-20151119220103-975617b05ea8
-	google.golang.org/grpc v0.0.0-20160317175043-d3ddb4469d5a // indirect
-	gopkg.in/check.v1 v1.0.0-20141024133853-64131543e789
-	gopkg.in/yaml.v2 v2.2.2
+	google.golang.org/grpc v1.21.0 // indirect
```
Comment (dmcgowan) [inline go.mod]: where is this grpc version change from?

File: .golangci.yml (hunk 3 above)
Comment (sayboras) [reply .golangci.yml]: Oh this one is coming from `example` config from golangci-lint.

However, it's making sense to keep the same as other repo (e.g. containerd). Let me make an update.

File: go.mod (hunk 4 above)
Comment (sayboras) [reply go.mod]: This is unintentional changes, I have reverted go.mod

**Comment by sayboras** [comment]
@dmcgowan the changes are cleaner now. Please help to take another look.

**Comment by sayboras** [comment]
@dmcgowan please let me know if you have any comments, happy to address.

File: .golangci.yml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    - vet
+    - unused
+    - misspell
+  disable:
+    - errcheck
+
+run:
+  deadline: 2m
+  skip-dirs:
+    - vendor
```
Comment (thaJeztah) [inline .golangci.yml]: nit: missing newline at end of file

Comment (sayboras) [reply .golangci.yml]: 💯

**Comment by sayboras** [comment]
> thanks! overall these changes look good, but some cleaning up might be good to not break `git bisect`; currently, the first commit switches to golangci-lint, and follow-up commits fix linking issues. That means that if someone would try a git bisect, the commit that switches to golangci-lint would "fail".

Just curious if it's ok for me to squash all commits into one :)

**Pull request merged**